File: java/src/main/java/com/svix/agentic_commerce_protocol/CheckPerformedType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum CheckPerformedType implements ToQueryParam {
    AVS("avs"),
    CVV("cvv"),
    ANI("ani"),
    AUTH0("auth0");
    private final String value;

    CheckPerformedType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/CheckPerformedType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class CheckPerformedType : ToQueryParam {
    @SerialName("avs") AVS,
    @SerialName("cvv") CVV,
    @SerialName("ani") ANI,
    @SerialName("auth0") AUTH0;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
